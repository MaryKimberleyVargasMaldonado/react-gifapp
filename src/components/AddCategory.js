import React, { useState } from "react";
import { ReactDOM } from "react";
import PropTypes from "prop-types";

export const AddCategory = (props) => {
	const [inputValue, setInputValue] = useState("Search a category:");

	// H A N D L E   I N P U T   C H A N G E
	const handleInputChange = (e) => {
		setInputValue(e.target.value);
		console.log("HandleInputChange llamado.");
	};

	//colocar el event a un form
	// H A N D L E   S U B M I T
	const handleSubmit = (e) => {
		e.preventDefault(); //prevenir comportamiento por defecto del formulario.
		console.log("Submit done", inputValue);

		if (inputValue.trim().length > 2) {
			props.setCategories((cate) => [inputValue, ...cate]);
			setInputValue("");
		}
	};

	return (
		//FRAGMENT
		<form onSubmit={handleSubmit}>
			<p>{inputValue}</p>
			<h2>{inputValue}</h2>
			<input type="text" value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

// P R O P E R T I E S
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
