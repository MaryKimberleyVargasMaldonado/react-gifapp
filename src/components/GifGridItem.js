import React from "react";
//import { GifGrid } from "./components/GiftGrid";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
	console.log({ title, url });
	return (
		<div className="card animate__animated animate__zoomInDown">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

// P R O P E R T I E S   -   P R O P T Y P E S
GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
	//id: PropTypes.string.isRequired,
};
