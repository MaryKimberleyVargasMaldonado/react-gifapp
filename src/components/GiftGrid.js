import React from "react";
import { useState, useEffect, Fragment } from "react/cjs/react.development";
//import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
	// U S E   F E T C H   G I F T S
	const { data: images, loading } = useFetchGifs(category);

	// I M A G E S
	/*
	const [images, setImages] = useState([]);


*/
	//getGifs();
	return (
		<Fragment>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			{loading && <p className="animate__animate animate__flash">Loading...</p>}
			<div className="card-grid">
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</Fragment>
	);
};

/*
			<ol>
				{images.map((img) => (
					<li key={img.id}>{img.title}</li>
				))}
			</ol>*/
