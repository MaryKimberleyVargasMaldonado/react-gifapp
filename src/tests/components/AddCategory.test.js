import { AddCategory } from "../../components/AddCategory";
import { shallow } from "enzyme";
import React from "react";
import "@testing-library/jest-dom";

describe("Pruebas en AddCategory.js", () => {
	const setCategories = jest.fn();
	let wrapper = shallow(<AddCategory setCategories={setCategories} />);

	// B E F O R E   E A C H
	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});

	// P R I M E R    T E S T
	test("1. <AddCategory /> mostrar correctamente: ", () => {
		expect(wrapper).toMatchSnapshot();
	});

	// S E G U N D O    T E S T
	test("2. La caja de texto debe cambiar: ", () => {
		const input = wrapper.find("input");
		const value = "Hola mundo";
		input.simulate("change", { target: { value } });
		expect(wrapper.find("p").text().trim()).toBe(value);
	});

	// T E R C E R   T E S T -  VÍDEO 95 PENDIENTE
	test("3. No debe postear la info con submit: ", () => {
		wrapper.find("form").simulate("submit", { preventDefault() {} });
		//expect(setCategories).not.toHaveBeenCalled();
	});

	// C U A R T O   T E S T
	test("4. Debe llamar el setCategories y limpiar la caja de texto: ", () => {
		//1. Simular el imput change.
		const value = "Husky";
		wrapper.find("input").simulate("change", { target: { value } });

		//2. Simular el submit.
		wrapper.find("form").simulate("submit", { preventDefault() {} });

		//3. setCategories been called.
		expect(setCategories).toHaveBeenCalled();
		expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

		//4. Valor de imput vacio.
		expect(wrapper.find("input").prop("value")).toBe("");
	});
}); //desc
