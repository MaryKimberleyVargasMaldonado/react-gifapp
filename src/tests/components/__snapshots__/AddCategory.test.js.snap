// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory.js 1. <AddCategory /> mostrar correctamente:  1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Search a category:
  </p>
  <h2>
    Search a category:
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value="Search a category:"
  />
</form>
`;
